src/common/components/nav/nav.component.js

```diff
@@ -37,6 +45,34 @@ class NavController{
     this.getNav().subscribe((structure) => {
       this.menuStructure = structure;
     });
+
+    this.subscription = this.sessionService.sessionSubject.subscribe( () => this.sessionChanged() );
+
+    this.templateUrl = this.$window.screen && this.$window.screen.width < 991 ? mobileTemplate.name : desktopTemplate.name;
```

Comment:
I'd assume Amazon does it cuz they have a completely separate mobile site... and they have a min width. This desktop menu looks broken on small sizes (especially the search bar) but that's not an issue for this PR.